internal/resource/config/oauth/oauth_access_token_manager_resource.go

```diff
@@ -526,60 +424,41 @@ func readOauthAccessTokenManagerResponse(ctx context.Context, r *client.AccessTo
+		diags.Append(respDiags...)
 
 		attributeContractValues := map[string]attr.Value{
 			"core_attributes":           attributeContractCoreAttributes,
 			"extended_attributes":       attributeContractExtendedAttributes,
 			"inherited":                 types.BoolPointerValue(attrContract.Inherited),
 			"default_subject_attribute": types.StringPointerValue(attrContract.DefaultSubjectAttribute),
 		}
-		state.AttributeContract, diags = types.ObjectValue(attributeContractTypes, attributeContractValues)
-		respDiags.Append(diags...)
+		state.AttributeContract, respDiags = types.ObjectValue(attributeContractTypes, attributeContractValues)
+		diags.Append(respDiags...)
 	}
 
 	// state.SelectionSettings
-	selectionSettingsAttrType := map[string]attr.Type{
-		"inherited":     basetypes.BoolType{},
-		"resource_uris": basetypes.ListType{ElemType: basetypes.StringType{}},
-	}
-
 	if r.SelectionSettings == nil {
 		state.SelectionSettings = types.ObjectNull(selectionSettingsAttrType)
 	} else {
-		state.SelectionSettings, diags = types.ObjectValueFrom(ctx, selectionSettingsAttrType, r.SelectionSettings)
-		respDiags.Append(diags...)
+		state.SelectionSettings, respDiags = types.ObjectValueFrom(ctx, selectionSettingsAttrType, r.SelectionSettings)
+		diags.Append(respDiags...)
 	}
 
 	// state.AccessControlSettings
-	accessControlSettingsAttrType := map[string]attr.Type{
-		"inherited":        basetypes.BoolType{},
-		"restrict_clients": basetypes.BoolType{},
-		"allowed_clients":  basetypes.ListType{ElemType: basetypes.ObjectType{AttrTypes: resourcelink.AttrType()}},
-	}
-
 	if r.AccessControlSettings == nil {
 		state.AccessControlSettings = types.ObjectNull(accessControlSettingsAttrType)
 	} else {
-		state.AccessControlSettings, diags = types.ObjectValueFrom(ctx, accessControlSettingsAttrType, r.AccessControlSettings)
-		respDiags.Append(diags...)
+		state.AccessControlSettings, respDiags = types.ObjectValueFrom(ctx, accessControlSettingsAttrType, r.AccessControlSettings)
+		diags.Append(respDiags...)
 	}
 
 	// state.SessionValidationSettings
-	sessionValidationSettingsAttrType := map[string]attr.Type{
-		"inherited":                       basetypes.BoolType{},
-		"include_session_id":              basetypes.BoolType{},
-		"check_valid_authn_session":       basetypes.BoolType{},
-		"check_session_revocation_status": basetypes.BoolType{},
-		"update_authn_session_activity":   basetypes.BoolType{},
-	}
-
 	if r.SessionValidationSettings == nil {
 		state.SessionValidationSettings = types.ObjectNull(sessionValidationSettingsAttrType)
 	} else {
-		state.SessionValidationSettings, diags = types.ObjectValueFrom(ctx, sessionValidationSettingsAttrType, r.SessionValidationSettings)
-		respDiags.Append(diags...)
+		state.SessionValidationSettings, respDiags = types.ObjectValueFrom(ctx, sessionValidationSettingsAttrType, r.SessionValidationSettings)
+		diags.Append(respDiags...)
```

Comment:
You are appending to diags this whole method and then returning respDiags. Should be returning the one we are appending to